cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/partitioned/PartitionedFileSetDataset.java

```diff
@@ -673,8 +759,8 @@ public String getOutputFormatClassName() {
   @Override
   public void onSuccess() throws DataSetException {
     String outputPath = FileSetArguments.getOutputPath(runtimeArguments);
-    // if there is no output path, the batch job probably would have failed
-    // we definitely can't do much here if we don't know the output path
+    // If there is no output path, it is either using DynamicPartitioner or the job would have failed.
+    // Either way, we can't do much here.
```

Comment:
do you think you should call files.onSuccess() here? It may not be doing anything today, but if in the future someone changes that, he may not know that he has to add the call here.